## DAY 2 :

# DATA ANALYSIS FOR WORLD HAPPINESS REPORT

- This is an analysis project to understand the world happiness report of countries

In [948]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

- pandas for data manipulation

- matplotlib and seaborn for visualization

## Loading Data 

In [949]:
df=pd.read_csv("world-happiness-report - world-happiness-report.csv")
df

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
0,Afghanistan,2008,3.724,7.370,0.451,50.80,0.718,0.168,0.882,0.518,0.258
1,Afghanistan,2009,4.402,7.540,0.552,51.20,0.679,0.190,0.850,0.584,0.237
2,Afghanistan,2010,4.758,7.647,0.539,51.60,0.600,0.121,0.707,0.618,0.275
3,Afghanistan,2011,3.832,7.620,0.521,51.92,0.496,0.162,0.731,0.611,0.267
4,Afghanistan,2012,3.783,7.705,0.521,52.24,0.531,0.236,0.776,0.710,0.268
...,...,...,...,...,...,...,...,...,...,...,...
1944,Zimbabwe,2016,3.735,7.984,0.768,54.40,0.733,-0.095,0.724,0.738,0.209
1945,Zimbabwe,2017,3.638,8.016,0.754,55.00,0.753,-0.098,0.751,0.806,0.224
1946,Zimbabwe,2018,3.616,8.049,0.775,55.60,0.763,-0.068,0.844,0.710,0.212
1947,Zimbabwe,2019,2.694,7.950,0.759,56.20,0.632,-0.064,0.831,0.716,0.235


## Inspecting Data 

In [950]:
df.head(5)

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
0,Afghanistan,2008,3.724,7.370,0.451,50.80,0.718,0.168,0.882,0.518,0.258
1,Afghanistan,2009,4.402,7.540,0.552,51.20,0.679,0.190,0.850,0.584,0.237
2,Afghanistan,2010,4.758,7.647,0.539,51.60,0.600,0.121,0.707,0.618,0.275
3,Afghanistan,2011,3.832,7.620,0.521,51.92,0.496,0.162,0.731,0.611,0.267
4,Afghanistan,2012,3.783,7.705,0.521,52.24,0.531,0.236,0.776,0.710,0.268


In [951]:
df.tail(5)

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
1944,Zimbabwe,2016,3.735,7.984,0.768,54.4,0.733,-0.095,0.724,0.738,0.209
1945,Zimbabwe,2017,3.638,8.016,0.754,55.0,0.753,-0.098,0.751,0.806,0.224
1946,Zimbabwe,2018,3.616,8.049,0.775,55.6,0.763,-0.068,0.844,0.710,0.212
1947,Zimbabwe,2019,2.694,7.950,0.759,56.2,0.632,-0.064,0.831,0.716,0.235
1948,Zimbabwe,2020,3.160,7.829,0.717,56.8,0.643,-0.009,0.789,0.703,0.346


In [952]:
df.sample(5)

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
1096,Mauritania,2015,3.923,8.542,0.875,56.10,0.447,0.055,0.715,0.820,0.194
441,Czech Republic,2012,6.334,10.426,0.912,68.38,0.740,-0.154,0.957,0.609,0.257
1757,Turkey,2010,5.490,9.906,0.795,64.40,0.515,-0.187,0.811,0.652,0.327
1084,Malta,2017,6.676,10.635,0.937,72.20,0.924,0.253,0.690,0.721,0.302
995,Libya,2012,5.754,9.842,0.855,62.66,0.712,-0.032,0.791,0.695,0.316


## Checking Data Types

In [953]:
df.dtypes

Country name                            str
year                                  int64
Life Ladder                         float64
Log GDP per capita                  float64
Social support                      float64
Healthy life expectancy at birth    float64
Freedom to make life choices        float64
Generosity                          float64
Perceptions of corruption           float64
Positive affect                     float64
Negative affect                     float64
dtype: object

In [954]:
df.columns

Index(['Country name', 'year', 'Life Ladder', 'Log GDP per capita',
       'Social support', 'Healthy life expectancy at birth',
       'Freedom to make life choices', 'Generosity',
       'Perceptions of corruption', 'Positive affect', 'Negative affect'],
      dtype='str')

In [955]:
df.shape

(1949, 11)

In [956]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1949 entries, 0 to 1948
Data columns (total 11 columns):
 #   Column                            Non-Null Count  Dtype  
---  ------                            --------------  -----  
 0   Country name                      1949 non-null   str    
 1   year                              1949 non-null   int64  
 2   Life Ladder                       1949 non-null   float64
 3   Log GDP per capita                1913 non-null   float64
 4   Social support                    1936 non-null   float64
 5   Healthy life expectancy at birth  1894 non-null   float64
 6   Freedom to make life choices      1917 non-null   float64
 7   Generosity                        1860 non-null   float64
 8   Perceptions of corruption         1839 non-null   float64
 9   Positive affect                   1927 non-null   float64
 10  Negative affect                   1933 non-null   float64
dtypes: float64(9), int64(1), str(1)
memory usage: 167.6 KB


In [957]:
df.duplicated().sum()

np.int64(0)

In [958]:
df.isnull().sum()

Country name                          0
year                                  0
Life Ladder                           0
Log GDP per capita                   36
Social support                       13
Healthy life expectancy at birth     55
Freedom to make life choices         32
Generosity                           89
Perceptions of corruption           110
Positive affect                      22
Negative affect                      16
dtype: int64

## Dataset Overview

- number of rows = 1949
- number of columns = 11
- data types: float = 9,integer = 1,object = 1
- missing values = none
- memory used = 167.6 + KB

## DAY 3 :

##  Data Cleaning

### cleaning numeric columns with characters

In [959]:
df[df["Log GDP per capita"].isnull()]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
423,Cuba,2006,5.418,NaN,0.970,68.44,0.281,NaN,NaN,0.647,0.277
436,Cyprus,2020,6.260,NaN,0.806,74.10,0.763,NaN,0.816,0.759,0.284
709,Hong Kong S.A.R. of China,2020,5.295,NaN,0.813,NaN,0.705,NaN,0.380,0.609,0.210
771,Iran,2018,4.278,NaN,0.674,66.00,0.603,NaN,0.703,0.553,0.493
772,Iran,2019,5.006,NaN,0.698,66.30,0.623,NaN,0.728,0.600,0.449
773,Iran,2020,4.865,NaN,0.757,66.60,0.600,NaN,0.710,0.582,0.470
918,Kosovo,2020,6.294,NaN,0.792,NaN,0.880,NaN,0.910,0.726,0.201
1087,Malta,2020,6.157,NaN,0.938,72.20,0.931,NaN,0.675,0.601,0.411
1280,North Cyprus,2012,5.463,NaN,0.871,NaN,0.693,NaN,0.855,0.709,0.405
1281,North Cyprus,2013,5.567,NaN,0.869,NaN,0.775,NaN,0.715,0.622,0.443


In [960]:
df[df["Country name"]=="Cuba"]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
423,Cuba,2006,5.418,NaN,0.97,68.44,0.281,NaN,NaN,0.647,0.277


In [961]:
df[df["Country name"]=="Somaliland region"]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
1562,Somaliland region,2009,4.991,NaN,0.880,NaN,0.746,NaN,0.513,0.819,0.112
1563,Somaliland region,2010,4.657,NaN,0.829,NaN,0.820,NaN,0.471,0.769,0.083
1564,Somaliland region,2011,4.931,NaN,0.788,NaN,0.858,NaN,0.357,0.749,0.122
1565,Somaliland region,2012,5.057,NaN,0.786,NaN,0.758,NaN,0.334,0.735,0.152


In [962]:
df[df["Country name"]=="Cyprus"]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
424,Cyprus,2006,6.238,10.566,0.878,71.44,0.836,0.018,0.712,0.829,0.253
425,Cyprus,2009,6.833,10.558,0.812,72.16,0.775,0.054,0.801,0.746,0.329
426,Cyprus,2010,6.387,10.551,0.822,72.40,0.755,0.073,0.833,0.786,0.296
427,Cyprus,2011,6.690,10.530,0.844,72.54,0.745,0.180,0.841,0.763,0.272
428,Cyprus,2012,6.181,10.479,0.767,72.68,0.725,0.098,0.871,0.755,0.369
429,Cyprus,2013,5.439,10.414,0.744,72.82,0.656,0.102,0.867,0.748,0.420
430,Cyprus,2014,5.627,10.406,0.770,72.96,0.715,0.060,0.868,0.737,0.397
431,Cyprus,2015,5.439,10.445,0.770,73.10,0.628,0.114,0.893,0.747,0.383
432,Cyprus,2016,5.795,10.506,0.786,73.30,0.756,-0.030,0.898,0.742,0.336
433,Cyprus,2017,6.062,10.539,0.819,73.50,0.812,0.043,0.851,0.784,0.301


In [963]:
df=df.dropna(subset=["Log GDP per capita"])

In [964]:
df.isnull().sum()

Country name                          0
year                                  0
Life Ladder                           0
Log GDP per capita                    0
Social support                       13
Healthy life expectancy at birth     37
Freedom to make life choices         32
Generosity                           53
Perceptions of corruption           109
Positive affect                      22
Negative affect                      16
dtype: int64

## DAY 4:

In [965]:
df[df["Social support"].isnull()]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
25,Algeria,2010,5.464,9.287,NaN,64.50,0.593,-0.205,0.618,NaN,NaN
112,Bahrain,2014,6.165,10.783,NaN,67.42,NaN,NaN,NaN,NaN,NaN
294,Canada,2007,7.482,10.739,NaN,71.66,0.930,0.249,0.406,0.872,0.257
466,Djibouti,2010,5.006,7.812,NaN,54.30,0.764,-0.058,0.597,NaN,NaN
925,Kuwait,2014,6.180,10.945,NaN,65.80,NaN,NaN,NaN,NaN,NaN
1164,Morocco,2010,4.383,8.746,NaN,63.50,0.663,-0.162,0.900,NaN,NaN
1310,Oman,2011,6.853,10.382,NaN,65.50,0.916,0.025,NaN,NaN,0.295
1423,Qatar,2010,6.850,11.520,NaN,66.70,NaN,0.104,NaN,NaN,NaN
1426,Qatar,2015,6.375,11.486,NaN,68.30,NaN,NaN,NaN,NaN,NaN
1741,Tunisia,2009,5.025,9.197,NaN,64.96,0.781,-0.119,0.722,NaN,NaN


In [966]:
df=df.dropna(subset=['Social support'])

In [967]:
df.isnull().sum()

Country name                          0
year                                  0
Life Ladder                           0
Log GDP per capita                    0
Social support                        0
Healthy life expectancy at birth     37
Freedom to make life choices         26
Generosity                           48
Perceptions of corruption           102
Positive affect                      11
Negative affect                       6
dtype: int64

In [968]:
df[df["Healthy life expectancy at birth"].isnull()]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
699,Hong Kong S.A.R. of China,2006,5.511,10.746,0.812,NaN,0.910,0.156,0.356,0.723,0.236
700,Hong Kong S.A.R. of China,2008,5.137,10.816,0.840,NaN,0.922,0.296,0.274,0.719,0.237
701,Hong Kong S.A.R. of China,2009,5.397,10.788,0.835,NaN,0.918,0.308,0.272,0.762,0.210
702,Hong Kong S.A.R. of China,2010,5.643,10.847,0.857,NaN,0.890,0.332,0.256,0.710,0.183
703,Hong Kong S.A.R. of China,2011,5.474,10.887,0.846,NaN,0.894,0.235,0.245,0.734,0.196
704,Hong Kong S.A.R. of China,2012,5.484,10.893,0.826,NaN,0.880,0.222,0.380,0.715,0.183
705,Hong Kong S.A.R. of China,2014,5.458,10.940,0.834,NaN,0.843,0.224,0.423,0.684,0.243
706,Hong Kong S.A.R. of China,2016,5.498,10.970,0.832,NaN,0.800,0.100,0.403,0.664,0.213
707,Hong Kong S.A.R. of China,2017,5.362,11.000,0.831,NaN,0.831,0.140,0.416,0.640,0.201
708,Hong Kong S.A.R. of China,2019,5.659,11.000,0.856,NaN,0.727,0.067,0.432,0.599,0.358


In [969]:
df[df["Country name"]=="Hong Kong S.A.R. of China"]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
699,Hong Kong S.A.R. of China,2006,5.511,10.746,0.812,NaN,0.910,0.156,0.356,0.723,0.236
700,Hong Kong S.A.R. of China,2008,5.137,10.816,0.840,NaN,0.922,0.296,0.274,0.719,0.237
701,Hong Kong S.A.R. of China,2009,5.397,10.788,0.835,NaN,0.918,0.308,0.272,0.762,0.210
702,Hong Kong S.A.R. of China,2010,5.643,10.847,0.857,NaN,0.890,0.332,0.256,0.710,0.183
703,Hong Kong S.A.R. of China,2011,5.474,10.887,0.846,NaN,0.894,0.235,0.245,0.734,0.196
704,Hong Kong S.A.R. of China,2012,5.484,10.893,0.826,NaN,0.880,0.222,0.380,0.715,0.183
705,Hong Kong S.A.R. of China,2014,5.458,10.940,0.834,NaN,0.843,0.224,0.423,0.684,0.243
706,Hong Kong S.A.R. of China,2016,5.498,10.970,0.832,NaN,0.800,0.100,0.403,0.664,0.213
707,Hong Kong S.A.R. of China,2017,5.362,11.000,0.831,NaN,0.831,0.140,0.416,0.640,0.201
708,Hong Kong S.A.R. of China,2019,5.659,11.000,0.856,NaN,0.727,0.067,0.432,0.599,0.358


In [970]:
df=df[~((df['Country name']=="Hong Kong S.A.R. of China") & (df['Healthy life expectancy at birth'].isna()))]

In [971]:
df[df["Country name"]=="Kosovo"]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
905,Kosovo,2007,5.104,8.928,0.848,NaN,0.381,0.144,0.894,0.655,0.237
906,Kosovo,2008,5.522,8.981,0.884,NaN,NaN,0.090,0.849,NaN,0.318
907,Kosovo,2009,5.891,9.008,0.830,NaN,0.506,0.201,0.968,0.598,0.169
908,Kosovo,2010,5.177,9.033,0.708,NaN,0.451,0.170,0.967,0.695,0.118
909,Kosovo,2011,4.860,9.067,0.759,NaN,0.589,0.004,0.919,0.696,0.124
910,Kosovo,2012,5.640,9.086,0.757,NaN,0.636,0.027,0.950,0.596,0.100
911,Kosovo,2013,6.126,9.113,0.721,NaN,0.568,0.115,0.935,0.692,0.203
912,Kosovo,2014,5.000,9.129,0.706,NaN,0.441,0.012,0.775,0.636,0.206
913,Kosovo,2015,5.077,9.182,0.805,NaN,0.561,0.181,0.851,0.753,0.180
914,Kosovo,2016,5.759,9.228,0.824,NaN,0.827,0.125,0.941,0.704,0.150


In [972]:
df=df[~((df['Country name']=="Kosovo") & (df['Healthy life expectancy at birth'].isna()))]

In [973]:
df[df["Country name"]=="Palestinian Territories"]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
1325,Palestinian Territories,2006,4.716,8.213,0.818,61.780,0.547,NaN,0.858,0.497,0.431
1326,Palestinian Territories,2007,4.151,8.218,0.712,61.897,0.365,-0.080,0.844,0.566,0.412
1327,Palestinian Territories,2008,4.386,8.276,0.666,62.015,0.358,-0.070,0.753,0.571,0.403
1328,Palestinian Territories,2009,4.470,8.329,0.738,62.132,0.468,-0.085,0.797,0.544,0.466
1329,Palestinian Territories,2010,4.703,8.383,0.822,62.250,0.504,-0.117,0.752,0.628,0.381
1330,Palestinian Territories,2011,4.751,8.474,0.751,NaN,0.522,-0.127,0.750,0.567,0.388
1331,Palestinian Territories,2012,4.647,8.531,0.782,NaN,0.542,-0.153,0.730,0.616,0.379
1332,Palestinian Territories,2013,4.844,8.489,0.761,NaN,0.454,-0.150,0.780,0.594,0.365
1333,Palestinian Territories,2014,4.722,8.457,0.775,NaN,0.657,-0.147,0.804,0.565,0.380
1334,Palestinian Territories,2015,4.695,8.480,0.766,NaN,0.556,-0.153,0.774,0.594,0.369


In [974]:
df[df["Country name"]=="Palestinian Territories"]["Healthy life expectancy at birth"].mean()

np.float64(62.014799999999994)

In [975]:
df.loc[(df["Country name"]=="Palestinian Territories") & (df["Healthy life expectancy at birth"].isna()),"Healthy life expectancy at birth"]=(62.014799999999994)

In [976]:
df[df["Country name"]=="Taiwan Province of China"]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
1669,Taiwan Province of China,2006,6.189,10.613,0.882,68.68,0.630,-0.030,0.846,0.814,0.094
1670,Taiwan Province of China,2008,5.548,10.606,0.830,69.14,0.642,-0.017,0.785,0.794,0.169
1671,Taiwan Province of China,2010,6.229,10.691,0.831,69.60,0.677,0.005,0.821,0.845,0.136
1672,Taiwan Province of China,2011,6.309,10.705,0.863,NaN,0.761,0.035,0.755,0.827,0.112
1673,Taiwan Province of China,2012,6.126,10.716,0.825,NaN,0.698,0.022,0.803,0.821,0.140
1674,Taiwan Province of China,2013,6.340,10.750,0.817,NaN,0.690,0.002,0.841,0.846,0.124
1675,Taiwan Province of China,2014,6.363,10.798,0.870,NaN,0.693,0.092,0.866,0.849,0.108
1676,Taiwan Province of China,2015,6.450,10.842,0.885,NaN,0.701,0.019,0.857,0.832,0.129
1677,Taiwan Province of China,2016,6.513,10.855,0.895,NaN,0.719,-0.049,0.811,0.833,0.108
1678,Taiwan Province of China,2017,6.359,10.871,0.891,NaN,0.760,-0.070,0.743,0.837,0.114


In [977]:

df[df["Country name"]=="Taiwan Province of China"]["Healthy life expectancy at birth"].mean()

np.float64(69.14)

In [978]:
df.loc[(df["Country name"]=="Taiwan Province of China") & (df["Healthy life expectancy at birth"].isna()),"Healthy life expectancy at birth"]=(69.14)

In [979]:
df.isnull().sum()

Country name                          0
year                                  0
Life Ladder                           0
Log GDP per capita                    0
Social support                        0
Healthy life expectancy at birth      0
Freedom to make life choices         25
Generosity                           48
Perceptions of corruption           102
Positive affect                      10
Negative affect                       6
dtype: int64

## DAY 5 :

In [980]:
df[df["Negative affect"].isnull()]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
872,Jordan,2018,4.639,9.196,0.800,66.8,0.762,-0.186,NaN,NaN,NaN
873,Jordan,2019,4.453,9.201,0.793,67.0,0.726,-0.165,NaN,NaN,NaN
874,Jordan,2020,4.094,9.150,0.709,67.2,0.779,-0.150,NaN,NaN,NaN
1062,Maldives,2018,5.198,9.826,0.913,70.6,0.855,0.024,NaN,NaN,NaN
1380,Philippines,2006,4.670,8.562,0.795,59.8,0.828,0.063,0.841,0.832,NaN
1753,Turkey,2005,4.719,9.809,0.820,62.6,0.623,NaN,0.877,0.557,NaN


In [981]:
df[df["Country name"]=="Jordan"]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
860,Jordan,2005,6.295,9.246,0.920,63.50,NaN,NaN,0.670,0.696,0.240
861,Jordan,2007,5.598,9.321,0.841,63.98,0.646,-0.112,0.664,0.683,0.240
862,Jordan,2008,4.930,9.343,0.766,64.22,NaN,-0.127,0.709,0.669,0.331
863,Jordan,2009,6.000,9.347,0.899,64.46,0.771,-0.075,0.739,0.645,0.265
864,Jordan,2010,5.570,9.317,0.918,64.70,0.788,-0.046,NaN,0.643,0.343
865,Jordan,2011,5.539,9.289,0.878,65.00,0.760,-0.143,NaN,0.612,0.260
866,Jordan,2012,5.132,9.261,0.829,65.30,0.693,-0.160,NaN,0.565,0.345
867,Jordan,2013,5.172,9.237,0.840,65.60,0.692,-0.117,NaN,0.684,0.286
868,Jordan,2014,5.333,9.222,0.816,65.90,0.729,-0.104,NaN,0.660,0.313
869,Jordan,2015,5.405,9.207,0.830,66.20,0.767,-0.046,NaN,0.690,0.305


In [982]:
df[df["Country name"]=="Maldives"]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
1062,Maldives,2018,5.198,9.826,0.913,70.6,0.855,0.024,NaN,NaN,NaN


In [983]:
df[df["Country name"]=="Philippines"]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
1380,Philippines,2006,4.670,8.562,0.795,59.8,0.828,0.063,0.841,0.832,NaN
1381,Philippines,2007,5.074,8.608,0.801,60.0,0.852,-0.022,0.880,0.784,0.378
1382,Philippines,2008,4.589,8.634,0.798,60.2,0.861,0.083,0.817,0.805,0.384
1383,Philippines,2009,4.880,8.632,0.775,60.4,0.874,0.004,0.805,0.846,0.311
1384,Philippines,2010,4.942,8.686,0.805,60.6,0.893,0.033,0.812,0.876,0.294
1385,Philippines,2011,4.994,8.707,0.789,60.8,0.883,0.073,0.783,0.851,0.358
1386,Philippines,2012,5.002,8.756,0.813,61.0,0.914,0.053,0.771,0.865,0.351
1387,Philippines,2013,4.977,8.805,0.846,61.2,0.907,0.021,0.756,0.799,0.332
1388,Philippines,2014,5.313,8.850,0.813,61.4,0.902,-0.015,0.787,0.813,0.334
1389,Philippines,2015,5.547,8.896,0.854,61.6,0.912,-0.051,0.755,0.805,0.351


In [984]:
df[df["Country name"]=="Turkey"]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
1753,Turkey,2005,4.719,9.809,0.820,62.60,0.623,NaN,0.877,0.557,NaN
1754,Turkey,2007,5.623,9.903,0.792,63.32,0.459,-0.178,0.800,0.651,0.395
1755,Turkey,2008,5.118,9.899,0.645,63.68,0.415,-0.189,0.785,0.614,0.345
1756,Turkey,2009,5.213,9.838,0.755,64.04,0.456,-0.227,0.853,0.598,0.316
1757,Turkey,2010,5.490,9.906,0.795,64.40,0.515,-0.187,0.811,0.652,0.327
1758,Turkey,2011,5.272,9.996,0.692,64.64,0.446,-0.242,0.649,0.621,0.380
1759,Turkey,2012,5.309,10.026,0.739,64.88,0.471,-0.216,0.702,0.645,0.335
1760,Turkey,2013,4.888,10.091,0.795,65.12,0.541,-0.229,0.698,0.635,0.392
1761,Turkey,2014,5.580,10.124,0.863,65.36,0.649,-0.024,0.764,0.483,0.377
1762,Turkey,2015,5.514,10.166,0.851,65.60,0.653,-0.016,0.806,0.460,0.382


In [985]:
df.isnull().sum()

Country name                          0
year                                  0
Life Ladder                           0
Log GDP per capita                    0
Social support                        0
Healthy life expectancy at birth      0
Freedom to make life choices         25
Generosity                           48
Perceptions of corruption           102
Positive affect                      10
Negative affect                       6
dtype: int64

In [986]:
df["Negative affect"]=df["Negative affect"].fillna(df.groupby("Country name")["Negative affect"].transform("mean"))

In [987]:
df.isnull().sum()

Country name                          0
year                                  0
Life Ladder                           0
Log GDP per capita                    0
Social support                        0
Healthy life expectancy at birth      0
Freedom to make life choices         25
Generosity                           48
Perceptions of corruption           102
Positive affect                      10
Negative affect                       1
dtype: int64

In [988]:
df[df["Positive affect"].isnull()]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
126,Bangladesh,2014,4.636,8.164,0.577,62.38,0.736,-0.098,0.789,NaN,0.231000
135,Belarus,2008,5.463,9.677,0.904,61.70,0.640,-0.220,0.696,NaN,0.246000
264,Cambodia,2007,4.156,7.829,0.675,56.10,0.819,0.116,0.879,NaN,0.320000
779,Iraq,2013,4.725,9.280,0.728,58.88,NaN,-0.050,0.710,NaN,0.554000
872,Jordan,2018,4.639,9.196,0.800,66.80,0.762,-0.186,NaN,NaN,0.302667
873,Jordan,2019,4.453,9.201,0.793,67.00,0.726,-0.165,NaN,NaN,0.302667
874,Jordan,2020,4.094,9.150,0.709,67.20,0.779,-0.150,NaN,NaN,0.302667
1062,Maldives,2018,5.198,9.826,0.913,70.60,0.855,0.024,NaN,NaN,NaN
1311,Pakistan,2005,5.225,8.218,0.591,54.20,0.630,NaN,0.844,NaN,0.237000
1813,United Arab Emirates,2013,6.621,11.001,0.864,66.46,0.936,NaN,NaN,NaN,0.291000


In [989]:
df[df["Country name"]=="Bangladesh"]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
118,Bangladesh,2006,4.319,7.783,0.672,59.02,0.612,0.068,0.786,0.600,0.321
119,Bangladesh,2007,4.607,7.839,0.514,59.44,0.605,0.040,0.806,0.635,0.313
120,Bangladesh,2008,5.052,7.886,0.467,59.86,0.606,-0.044,0.802,0.725,0.232
121,Bangladesh,2009,5.083,7.924,0.528,60.28,0.631,-0.075,0.776,0.670,0.223
122,Bangladesh,2010,4.858,7.967,0.549,60.70,0.659,-0.016,0.774,0.629,0.292
123,Bangladesh,2011,4.986,8.018,0.606,61.12,0.838,-0.069,0.757,0.685,0.235
124,Bangladesh,2012,4.724,8.070,0.582,61.54,0.668,-0.034,0.765,0.714,0.183
125,Bangladesh,2013,4.660,8.116,0.530,61.96,0.742,-0.016,0.743,0.619,0.246
126,Bangladesh,2014,4.636,8.164,0.577,62.38,0.736,-0.098,0.789,NaN,0.231
127,Bangladesh,2015,4.633,8.216,0.601,62.80,0.815,-0.068,0.721,0.635,0.226


In [990]:
df[df["Country name"]=="Belarus"]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
133,Belarus,2006,5.658,9.489,0.918,61.10,0.707,-0.246,0.708,0.605,0.269
134,Belarus,2007,5.617,9.576,0.858,61.40,0.667,-0.225,0.695,0.596,0.235
135,Belarus,2008,5.463,9.677,0.904,61.70,0.640,-0.220,0.696,NaN,0.246
136,Belarus,2009,5.564,9.681,0.908,62.00,0.679,-0.203,0.676,0.566,0.223
137,Belarus,2010,5.526,9.758,0.918,62.30,0.700,-0.163,0.706,0.567,0.208
138,Belarus,2011,5.225,9.812,0.910,62.88,0.656,-0.168,0.672,0.521,0.249
139,Belarus,2012,5.749,9.830,0.902,63.46,0.645,-0.217,0.657,0.523,0.181
140,Belarus,2013,5.876,9.839,0.923,64.04,0.723,-0.177,0.653,0.609,0.206
141,Belarus,2014,5.812,9.856,0.880,64.62,0.647,-0.048,0.682,0.619,0.209
142,Belarus,2015,5.719,9.815,0.924,65.20,0.623,-0.091,0.669,0.584,0.184


In [991]:
df[df["Country name"]=="Cambodia"]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
263,Cambodia,2006,3.569,7.746,0.793,55.30,NaN,0.255,0.829,0.719,0.341
264,Cambodia,2007,4.156,7.829,0.675,56.10,0.819,0.116,0.879,NaN,0.320
265,Cambodia,2008,4.462,7.879,0.619,56.90,0.914,0.045,0.888,0.739,0.335
266,Cambodia,2009,4.111,7.865,0.818,57.70,0.937,0.152,0.965,0.796,0.188
267,Cambodia,2010,4.141,7.907,0.697,58.50,0.940,0.350,0.896,0.774,0.422
268,Cambodia,2011,4.161,7.960,0.716,58.88,0.927,0.418,0.775,0.799,0.308
269,Cambodia,2012,3.899,8.014,0.606,59.26,0.956,0.247,0.890,0.820,0.352
270,Cambodia,2013,3.674,8.068,0.651,59.64,0.941,0.164,0.812,0.792,0.440
271,Cambodia,2014,3.883,8.121,0.693,60.02,0.938,0.239,0.843,0.783,0.482
272,Cambodia,2015,4.162,8.173,0.729,60.40,0.956,0.210,0.825,0.813,0.399


In [992]:
df["Positive affect"]=df["Positive affect"].fillna(df.groupby("Country name")["Positive affect"].transform("mean"))

In [993]:
df.isnull().sum()

Country name                          0
year                                  0
Life Ladder                           0
Log GDP per capita                    0
Social support                        0
Healthy life expectancy at birth      0
Freedom to make life choices         25
Generosity                           48
Perceptions of corruption           102
Positive affect                       1
Negative affect                       1
dtype: int64

In [994]:
df[df["Positive affect"].isnull()]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
1062,Maldives,2018,5.198,9.826,0.913,70.6,0.855,0.024,NaN,NaN,NaN


In [995]:
df=df.dropna(subset=["Positive affect"])

In [996]:
df.isnull().sum()

Country name                          0
year                                  0
Life Ladder                           0
Log GDP per capita                    0
Social support                        0
Healthy life expectancy at birth      0
Freedom to make life choices         25
Generosity                           48
Perceptions of corruption           101
Positive affect                       0
Negative affect                       0
dtype: int64

In [997]:
df[df["Freedom to make life choices"].isnull()]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
28,Algeria,2014,6.355,9.335,0.818,65.14,NaN,NaN,NaN,0.626,0.177
29,Algeria,2016,5.341,9.362,0.749,65.50,NaN,NaN,NaN,0.661,0.377
263,Cambodia,2006,3.569,7.746,0.793,55.30,NaN,0.255,0.829,0.719,0.341
342,China,2006,4.560,8.696,0.747,66.88,NaN,NaN,NaN,0.809,0.170
343,China,2007,4.863,8.824,0.811,67.06,NaN,-0.176,NaN,0.817,0.159
350,China,2014,5.196,9.386,0.820,68.24,NaN,-0.217,NaN,0.854,0.112
351,China,2015,5.304,9.449,0.794,68.40,NaN,-0.244,NaN,0.809,0.171
352,China,2016,5.325,9.510,0.742,68.70,NaN,-0.228,NaN,0.826,0.146
500,Egypt,2008,4.632,9.186,0.738,59.88,NaN,-0.087,0.914,0.683,0.301
779,Iraq,2013,4.725,9.280,0.728,58.88,NaN,-0.050,0.710,0.528,0.554


In [998]:
df[df["Country name"]=="Algeria"]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
26,Algeria,2011,5.317,9.297,0.810,64.66,0.530,-0.181,0.638,0.550,0.255
27,Algeria,2012,5.605,9.311,0.839,64.82,0.587,-0.172,0.690,0.604,0.230
28,Algeria,2014,6.355,9.335,0.818,65.14,NaN,NaN,NaN,0.626,0.177
29,Algeria,2016,5.341,9.362,0.749,65.50,NaN,NaN,NaN,0.661,0.377
30,Algeria,2017,5.249,9.354,0.807,65.70,0.437,-0.167,0.700,0.642,0.289
31,Algeria,2018,5.043,9.348,0.799,65.90,0.583,-0.146,0.759,0.591,0.293
32,Algeria,2019,4.745,9.337,0.803,66.10,0.385,0.005,0.741,0.585,0.215


In [999]:
df[df["Country name"]=="China"]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
342,China,2006,4.560,8.696,0.747,66.88,NaN,NaN,NaN,0.809,0.170
343,China,2007,4.863,8.824,0.811,67.06,NaN,-0.176,NaN,0.817,0.159
344,China,2008,4.846,8.911,0.748,67.24,0.853,-0.092,NaN,0.817,0.147
345,China,2009,4.454,8.996,0.798,67.42,0.771,-0.160,NaN,0.786,0.162
346,China,2010,4.653,9.092,0.768,67.60,0.805,-0.133,NaN,0.765,0.158
347,China,2011,5.037,9.179,0.787,67.76,0.824,-0.186,NaN,0.820,0.134
348,China,2012,5.095,9.249,0.788,67.92,0.808,-0.185,NaN,0.821,0.159
349,China,2013,5.241,9.319,0.778,68.08,0.805,-0.158,NaN,0.836,0.142
350,China,2014,5.196,9.386,0.820,68.24,NaN,-0.217,NaN,0.854,0.112
351,China,2015,5.304,9.449,0.794,68.40,NaN,-0.244,NaN,0.809,0.171


In [1000]:
df['Freedom to make life choices'] = df['Freedom to make life choices'].fillna(
    df.groupby('Country name')['Freedom to make life choices'].transform('mean')
)

In [1001]:
df.isnull().sum()

Country name                          0
year                                  0
Life Ladder                           0
Log GDP per capita                    0
Social support                        0
Healthy life expectancy at birth      0
Freedom to make life choices          0
Generosity                           48
Perceptions of corruption           101
Positive affect                       0
Negative affect                       0
dtype: int64

In [1002]:
df[df["Generosity"].isnull()]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
28,Algeria,2014,6.355,9.335,0.818,65.14,0.504400,NaN,NaN,0.626000,0.177000
29,Algeria,2016,5.341,9.362,0.749,65.50,0.504400,NaN,NaN,0.661000,0.377000
66,Australia,2005,7.341,10.659,0.968,71.40,0.935000,NaN,0.390,0.843000,0.238000
110,Bahrain,2012,5.027,10.716,0.911,66.86,0.682000,NaN,0.438,0.589000,0.381000
111,Bahrain,2013,6.690,10.757,0.884,67.14,0.809000,NaN,0.525,0.768000,0.306000
147,Belgium,2005,7.262,10.745,0.935,69.90,0.924000,NaN,0.598,0.796000,0.260000
218,Brazil,2005,6.637,9.438,0.883,63.30,0.882000,NaN,0.745,0.818000,0.302000
224,Brazil,2012,6.660,9.647,0.890,65.02,0.849000,NaN,0.623,0.755000,0.350000
342,China,2006,4.560,8.696,0.747,66.88,0.845700,NaN,NaN,0.809000,0.170000
437,Czech Republic,2005,6.439,10.324,0.919,67.00,0.865000,NaN,0.901,0.723000,0.258000


In [1003]:
df[df["Country name"]=="Turkey"]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
1753,Turkey,2005,4.719,9.809,0.820,62.60,0.623,NaN,0.877,0.557,0.365071
1754,Turkey,2007,5.623,9.903,0.792,63.32,0.459,-0.178,0.800,0.651,0.395000
1755,Turkey,2008,5.118,9.899,0.645,63.68,0.415,-0.189,0.785,0.614,0.345000
1756,Turkey,2009,5.213,9.838,0.755,64.04,0.456,-0.227,0.853,0.598,0.316000
1757,Turkey,2010,5.490,9.906,0.795,64.40,0.515,-0.187,0.811,0.652,0.327000
1758,Turkey,2011,5.272,9.996,0.692,64.64,0.446,-0.242,0.649,0.621,0.380000
1759,Turkey,2012,5.309,10.026,0.739,64.88,0.471,-0.216,0.702,0.645,0.335000
1760,Turkey,2013,4.888,10.091,0.795,65.12,0.541,-0.229,0.698,0.635,0.392000
1761,Turkey,2014,5.580,10.124,0.863,65.36,0.649,-0.024,0.764,0.483,0.377000
1762,Turkey,2015,5.514,10.166,0.851,65.60,0.653,-0.016,0.806,0.460,0.382000


In [1004]:
df[df["Country name"]=="Spain"]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
1600,Spain,2005,7.153,10.546,0.961,71.50,0.916,NaN,0.777,0.776,0.241
1601,Spain,2007,6.995,10.587,0.957,72.06,0.782,-0.093,0.784,0.763,0.264
1602,Spain,2008,7.294,10.579,0.948,72.34,0.834,-0.150,0.683,0.772,0.260
1603,Spain,2009,6.199,10.532,0.929,72.62,0.749,-0.127,0.798,0.752,0.336
1604,Spain,2010,6.188,10.529,0.950,72.90,0.796,-0.138,0.840,0.724,0.322
1605,Spain,2011,6.518,10.518,0.944,73.02,0.819,-0.122,0.846,0.737,0.356
1606,Spain,2012,6.291,10.487,0.937,73.14,0.755,-0.059,0.844,0.749,0.366
1607,Spain,2013,6.150,10.476,0.929,73.26,0.759,-0.101,0.916,0.696,0.372
1608,Spain,2014,6.456,10.492,0.948,73.38,0.738,-0.028,0.854,0.716,0.335
1609,Spain,2015,6.381,10.531,0.956,73.50,0.732,-0.072,0.822,0.732,0.285


In [1005]:
df['Generosity'] = df['Generosity'].fillna(
    df.groupby('Country name')['Generosity'].transform('mean')
)

In [1006]:
df.isnull().sum()

Country name                          0
year                                  0
Life Ladder                           0
Log GDP per capita                    0
Social support                        0
Healthy life expectancy at birth      0
Freedom to make life choices          0
Generosity                            0
Perceptions of corruption           101
Positive affect                       0
Negative affect                       0
dtype: int64

In [1007]:
df[df["Perceptions of corruption"].isnull()]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
28,Algeria,2014,6.355,9.335,0.818,65.14,0.504400,-0.1322,NaN,0.626,0.177
29,Algeria,2016,5.341,9.362,0.749,65.50,0.504400,-0.1322,NaN,0.661,0.377
113,Bahrain,2015,6.007,10.785,0.853,67.70,0.850000,0.1120,NaN,0.716,0.303
114,Bahrain,2016,6.170,10.781,0.863,68.10,0.889000,0.0880,NaN,0.787,0.283
115,Bahrain,2017,6.227,10.771,0.876,68.50,0.906000,0.1360,NaN,0.814,0.290
...,...,...,...,...,...,...,...,...,...,...,...
1902,Vietnam,2014,5.085,8.716,0.792,67.14,0.882455,0.0000,NaN,0.701,0.241
1903,Vietnam,2015,5.076,8.770,0.849,67.30,0.882455,0.0860,NaN,0.642,0.232
1908,Yemen,2007,4.477,8.214,0.825,53.40,0.673000,0.0110,NaN,0.592,0.379
1916,Yemen,2016,3.826,7.715,0.775,55.10,0.533000,-0.1510,NaN,0.469,0.228


In [1008]:
df[df["Country name"]=="Bahrain"]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
107,Bahrain,2009,5.701,10.709,0.904,65.94,0.896,0.037000,0.506,0.764,0.422
108,Bahrain,2010,5.937,10.706,0.877,66.30,0.862,-0.001000,0.715,0.685,0.423
109,Bahrain,2011,4.824,10.696,0.908,66.58,0.870,-0.051000,0.583,0.544,0.514
110,Bahrain,2012,5.027,10.716,0.911,66.86,0.682,0.062625,0.438,0.589,0.381
111,Bahrain,2013,6.690,10.757,0.884,67.14,0.809,0.062625,0.525,0.768,0.306
113,Bahrain,2015,6.007,10.785,0.853,67.70,0.850,0.112000,NaN,0.716,0.303
114,Bahrain,2016,6.170,10.781,0.863,68.10,0.889,0.088000,NaN,0.787,0.283
115,Bahrain,2017,6.227,10.771,0.876,68.50,0.906,0.136000,NaN,0.814,0.290
116,Bahrain,2019,7.098,10.715,0.878,69.30,0.907,0.048000,NaN,0.762,0.317
117,Bahrain,2020,6.173,10.620,0.848,69.70,0.945,0.132000,NaN,0.790,0.297


In [1009]:
df[df["Country name"]=="Yemen"]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
1908,Yemen,2007,4.477,8.214,0.825,53.4,0.673,0.011,NaN,0.592,0.379
1909,Yemen,2009,4.809,8.278,0.756,54.0,0.644,-0.052,0.832,0.583,0.374
1910,Yemen,2010,4.350,8.453,0.727,54.3,0.659,-0.104,0.853,0.582,0.308
1911,Yemen,2011,3.746,8.336,0.663,54.3,0.638,-0.173,0.754,0.503,0.285
1912,Yemen,2012,4.061,8.236,0.682,54.3,0.706,-0.171,0.793,0.502,0.263
1913,Yemen,2013,4.218,8.242,0.694,54.3,0.543,-0.179,0.885,0.558,0.266
1914,Yemen,2014,3.968,8.117,0.638,54.3,0.664,-0.157,0.885,0.611,0.276
1915,Yemen,2015,2.983,7.858,0.669,54.3,0.610,-0.139,0.829,0.507,0.321
1916,Yemen,2016,3.826,7.715,0.775,55.1,0.533,-0.151,NaN,0.469,0.228
1917,Yemen,2017,3.254,7.578,0.790,55.9,0.595,-0.147,NaN,0.455,0.295


In [1010]:
df[df["Country name"]=="Vietnam"]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
1894,Vietnam,2006,5.294,8.335,0.888,65.86,0.886000,0.015,NaN,0.682,0.204
1895,Vietnam,2007,5.422,8.394,0.856,66.02,0.918000,0.089,0.754,0.588,0.206
1896,Vietnam,2008,5.480,8.440,0.805,66.18,0.889000,0.201,0.789,0.665,0.218
1897,Vietnam,2009,5.304,8.483,0.815,66.34,0.834000,-0.062,0.838,0.583,0.190
1898,Vietnam,2010,5.296,8.535,0.787,66.50,0.831000,-0.006,0.743,0.685,0.216
1899,Vietnam,2011,5.767,8.585,0.898,66.66,0.818000,0.105,0.742,0.532,0.193
1900,Vietnam,2012,5.535,8.626,0.775,66.82,0.856000,-0.110,0.815,0.615,0.221
1901,Vietnam,2013,5.023,8.668,0.759,66.98,0.920000,-0.027,0.771,0.718,0.165
1902,Vietnam,2014,5.085,8.716,0.792,67.14,0.882455,0.000,NaN,0.701,0.241
1903,Vietnam,2015,5.076,8.770,0.849,67.30,0.882455,0.086,NaN,0.642,0.232


In [1011]:
df['Perceptions of corruption'] = df['Perceptions of corruption'].fillna(
    df.groupby('Country name')['Perceptions of corruption'].transform('mean')
)

In [1012]:
df.isnull().sum()

Country name                         0
year                                 0
Life Ladder                          0
Log GDP per capita                   0
Social support                       0
Healthy life expectancy at birth     0
Freedom to make life choices         0
Generosity                           0
Perceptions of corruption           25
Positive affect                      0
Negative affect                      0
dtype: int64

In [1013]:
df[df["Perceptions of corruption"].isnull()]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
342,China,2006,4.560,8.696,0.747,66.88,0.8457,-0.170643,NaN,0.809,0.170
343,China,2007,4.863,8.824,0.811,67.06,0.8457,-0.176000,NaN,0.817,0.159
344,China,2008,4.846,8.911,0.748,67.24,0.8530,-0.092000,NaN,0.817,0.147
345,China,2009,4.454,8.996,0.798,67.42,0.7710,-0.160000,NaN,0.786,0.162
346,China,2010,4.653,9.092,0.768,67.60,0.8050,-0.133000,NaN,0.765,0.158
347,China,2011,5.037,9.179,0.787,67.76,0.8240,-0.186000,NaN,0.820,0.134
348,China,2012,5.095,9.249,0.788,67.92,0.8080,-0.185000,NaN,0.821,0.159
349,China,2013,5.241,9.319,0.778,68.08,0.8050,-0.158000,NaN,0.836,0.142
350,China,2014,5.196,9.386,0.820,68.24,0.8457,-0.217000,NaN,0.854,0.112
351,China,2015,5.304,9.449,0.794,68.40,0.8457,-0.244000,NaN,0.809,0.171


In [1014]:
df[df["Country name"]=="China"]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
342,China,2006,4.560,8.696,0.747,66.88,0.8457,-0.170643,NaN,0.809,0.170
343,China,2007,4.863,8.824,0.811,67.06,0.8457,-0.176000,NaN,0.817,0.159
344,China,2008,4.846,8.911,0.748,67.24,0.8530,-0.092000,NaN,0.817,0.147
345,China,2009,4.454,8.996,0.798,67.42,0.7710,-0.160000,NaN,0.786,0.162
346,China,2010,4.653,9.092,0.768,67.60,0.8050,-0.133000,NaN,0.765,0.158
347,China,2011,5.037,9.179,0.787,67.76,0.8240,-0.186000,NaN,0.820,0.134
348,China,2012,5.095,9.249,0.788,67.92,0.8080,-0.185000,NaN,0.821,0.159
349,China,2013,5.241,9.319,0.778,68.08,0.8050,-0.158000,NaN,0.836,0.142
350,China,2014,5.196,9.386,0.820,68.24,0.8457,-0.217000,NaN,0.854,0.112
351,China,2015,5.304,9.449,0.794,68.40,0.8457,-0.244000,NaN,0.809,0.171


In [1015]:
df[df["Country name"]=="Turkmenistan"]

,Country name,year,Life Ladder,Log GDP per capita,Social support,Healthy life expectancy at birth,Freedom to make life choices,Generosity,Perceptions of corruption,Positive affect,Negative affect
1768,Turkmenistan,2009,6.568,8.989,0.924,59.44,0.777,-0.102,NaN,0.781,0.152
1769,Turkmenistan,2011,5.792,9.182,0.964,60.04,0.777,0.018,NaN,0.639,0.122
1770,Turkmenistan,2012,5.464,9.269,0.946,60.28,0.786,-0.123,NaN,0.584,0.117
1771,Turkmenistan,2013,5.392,9.348,0.846,60.52,0.705,-0.071,NaN,0.599,0.160
1772,Turkmenistan,2014,5.787,9.427,0.909,60.76,0.805,0.032,NaN,0.695,0.154
1773,Turkmenistan,2015,5.791,9.472,0.960,61.00,0.701,0.093,NaN,0.705,0.301
1774,Turkmenistan,2016,5.887,9.515,0.929,61.40,0.749,0.005,NaN,0.636,0.255
1775,Turkmenistan,2017,5.229,9.561,0.908,61.80,0.720,0.066,NaN,0.521,0.350
1776,Turkmenistan,2018,4.621,9.605,0.984,62.20,0.858,0.260,NaN,0.612,0.189
1777,Turkmenistan,2019,5.474,9.651,0.982,62.60,0.892,0.285,NaN,0.510,0.183


In [1016]:
df=df.dropna(subset=["Perceptions of corruption"])

In [1017]:
df.isnull().sum()

Country name                        0
year                                0
Life Ladder                         0
Log GDP per capita                  0
Social support                      0
Healthy life expectancy at birth    0
Freedom to make life choices        0
Generosity                          0
Perceptions of corruption           0
Positive affect                     0
Negative affect                     0
dtype: int64

In [1018]:
df.to_csv("cleaned_data.csv")